v-next/hardhat/src/internal/builtin-plugins/solidity/index.ts

```diff
@@ -28,6 +29,12 @@ const hardhatPlugin: HardhatPlugin = {
         name: "quiet",
         description: "Makes the compilation process less verbose",
       })
+      .addOption({
+        name: "defaultBuildProfile",
```

Comment:
Can this change to `buildProfile`?

Reply:
Or maybe `deploymenttBuildProfile`?

Reply:
Three changes needed:

1. change `globalOptions.buildProfile` to default to undefined
2. change compile task option `defaultBuildProfile` to default to "default"
3. change solidity test task to set its build profile as either the globalOption.buildProfile or "default"